Added compatibility with correction matrix

NehaBari opened the pull request:
1. Pulled changes from upstream to forked repo. 
2. Changed line 234 in _gmlvq.py to accept relevance correction as a parameter.

Commit:
Added compatibility with correction matrix

NehaBari committed:
fixed correction matrix

rickvanveen commented:
@NehaBari Good work! I changed it to a "draft" pull request, because some things need to happen before it can be accepted: 

The codecov checks fail, this means that the test coverage of the project goes down and the method should be tested. The tests are located in the module's test folder (i.e., sklvq/models/tests). In the `test_gmlvq.py` file you should make a function called `test_relevance_correction` and add a simple example for which you know the answer (unit eigenvectors for example). See other test for inspiration. 

Secondly, could you remove the old code (instead of commenting it out). We don't need the old (wrong) solution in the comments when using version control :-)

Thirdly, we should think about including an example in the documentation, i.e., an example that will be shown in [sklvq.readthedocs.io](https://sklvq.readthedocs.io). The examples that you see there are located [here](https://github.com/rickvanveen/sklvq/tree/fr-51-null-space-correction/examples).

NehaBari commented:
Dear Rick,

Thankyou for the detailed feedback. Sounds good. I will start  incorporating them.

NehaBari committed:
(AB)T= BTAT

Given matrices A and B so that the operations can be pre-formed

properties of a transpose

NehaBari committed:
(AB)T= BTAT

Given matrices A and B so that the operations can be pre-formed
properties of a transpose

removed some comments

rickvanveen commented on sklvq/models/_gmlvq.py:
```diff
@@ -386,8 +386,8 @@ def _normalize_omega(omega: np.ndarray) -> None:
-        np.matmul(omega, self.relevance_correction.T, out=omega)
-        #np.matmul(omega.T, self.relevance_correction, out=omega.T)
+        np.matmul(self.relevance_correction,omega,out=omega.T)
+        # will this still keep omega as a row matrix when there is a omega.T in out???????????????????????????????????????????????????        
```
Don't know. You could test this. Are you certain the transposes are now all correct? You changed them again. It would also depend on the correction matrix, are you sure that is correct?

rickvanveen commented on sklvq/models/_gmlvq.py:
```diff
@@ -385,7 +386,7 @@ def _normalize_omega(omega: np.ndarray) -> None:
     def _correct_omega(self, omega: np.ndarray) -> None:
         # Note matmul is faster for larger matrices, for smaller dot is faster.
-        np.matmul(omega.T, self.relevance_correction, out=omega.T)
+        np.matmul(self.relevance_correction,omega,out=omega.T)       
```
Are you sure this is correct and you did not change the way the correction matrix was defined?
